### PR TITLE
Fixes #RHINENG-9928 - upgrade werkzeug & flask

### PR DESCRIPTION
**Pull request by kgaikwad**
## PR Title :boom:

Please title this PR with a summary of the change, along with the JIRA card number.

Suggested formats: 

1. Fixes/Refs #RHIROS-XXX - Title
2. RHIROS-XXX Title 

Feel free to remove this section from PR description once done.

## Why do we need this change? :thought_balloon:

Please include the __context of this change__ here.

## Documentation update? :memo:

- [ ] Yes
- [ ] No

## Security Checklist :lock:

Upon raising this PR please go through [RedHatInsights/secure-coding-checklist](https://github.com/RedHatInsights/secure-coding-checklist)

## :guardsman: Checklist :dart:

- [ ] Bugfix
- [ ] New Feature
- [ ] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [ ] Dependency Added
- [ ] DB Migration Added

## Additional :mega:

With werkzeug, upgrade flask version. Please check below failures/warnings observed in test pipeline against werkzeug upgrade:

```
- tests/test_api_endpoints.py:665
  /home/kavita/projects_dir/cloud_services/ros-backend/tests/test_api_endpoints.py:665: PytestUnknownMarkWarning: Unknown pytest.mark.no_suggestions - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.no_suggestions


- tests/test_api_endpoints.py:769
  /home/kavita/projects_dir/cloud_services/ros-backend/tests/test_api_endpoints.py:769: PytestUnknownMarkWarning: Unknown pytest.mark.generate_suggestions - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.generate_suggestions

- ../../../.cache/pypoetry/virtualenvs/ros-backend-Oq-9_0IR-py3.11/lib/python3.11/site-packages/flask_unleash/__init__.py:3
../../../.cache/pypoetry/virtualenvs/ros-backend-Oq-9_0IR-py3.11/lib/python3.11/site-packages/flask_unleash/__init__.py:3
  /home/kavita/.cache/pypoetry/virtualenvs/ros-backend-Oq-9_0IR-py3.11/lib/python3.11/site-packages/flask_unleash/__init__.py:3: DeprecationWarning: '_app_ctx_stack' is deprecated and will be removed in Flask 2.3.
    from flask import current_app, _app_ctx_stack  # type: ignore

Note: This needs upgrade of flask_unleash - https://github.com/Unleash/Flask-Unleash/issues/19

- tests/test_api_endpoints.py: 46 warnings
  /home/kavita/.cache/pypoetry/virtualenvs/ros-backend-Oq-9_0IR-py3.11/lib/python3.11/site-packages/werkzeug/datastructures/headers.py:297: DeprecationWarning: Passing bytes as a header value is deprecated and will not be supported in Werkzeug 3.0.
    _value = _str_header_value(_value)
    
- tests/test_api_endpoints.py: 41 warnings
  /home/kavita/.cache/pypoetry/virtualenvs/ros-backend-Oq-9_0IR-py3.11/lib/python3.11/site-packages/flask/testing.py:116: DeprecationWarning: The '__version__' attribute is deprecated and will be removed in Werkzeug 3.1. Use feature detection or 'importlib.metadata.version("werkzeug")' instead.
    "HTTP_USER_AGENT": f"werkzeug/{werkzeug.__version__}"
    
- Multiple failures while running tests:
 FAILED tests/test_api_endpoints.py::test_get_suggested_instance_details_filtered_by_os - UnicodeDecodeError: 'utf-8' codec can't decode byte 0x91 in position 4: invalid start byte
 
 Note: Read about issue https://github.com/pallets/werkzeug/issues/2602
```
    
Close #429 after this PR change in main branch

### CODE REVIEW

**Comment by kgaikwad** [comment]
\retest

File: tests/fixtures/db_fixtures.py [hunk 1]
```diff
@@ -31,7 +32,7 @@ def auth_token():
             }
         }
     }
-    auth_token = b64encode(json.dumps(identity).encode('utf-8'))
+    auth_token = codecs.decode(b64encode(json.dumps(identity).encode('utf-8')))
```
Comment (upadhyeammit) [inline tests/fixtures/db_fixtures.py:35]: I think we don't need codecs here, I am able to convert bytes to string using,
```suggestion
    auth_token = b64encode(json.dumps(identity).encode('utf-8')).decode()
```

File: tests/test_api_endpoints.py [hunk 2]
```diff
@@ -329,7 +330,9 @@ def test_should_check_authorization_for_rating_request(
             }
         }
     }
-    auth_token_for_t1 = b64encode(json.dumps(identity).encode('utf-8'))
+    auth_token_for_t1 = codecs.decode(
+        b64encode(json.dumps(identity).encode('utf-8'))
+    )
```
Comment (upadhyeammit) [inline tests/test_api_endpoints.py:335]: same here as well,
```suggestion
    auth_token_for_t1 = b64encode(json.dumps(identity).encode('utf-8')).decode()
```

**Review by upadhyeammit — CHANGES_REQUESTED** [review]

File: tests/fixtures/db_fixtures.py (hunk 1 above)
Comment (kgaikwad) [reply tests/fixtures/db_fixtures.py]: +1 Thank you for pointing this out. Corrected at both places.

**Review by upadhyeammit — APPROVED** [review]

**Pull request merged**